src/nvrtc_util.cc

```diff
@@ -201,6 +201,9 @@ float const FLT_MIN = 1.175494350822287507969e-38f;
       }
 
       init_done.v = 1;
+      char dev_name[50];
```

Comment:
this should follow existing patterns for getting strings from the cuda driver API; see get_plat_tag() -- use a string, do error checking. once that is done, it might make sense to factor out a 'get string value from cuda driver api (template) function', so that snippet isn't duplicated.


also, you probably shouldn't unconditionally print info to stdout -- note how (all?) other printouts are guarded by flags/booleans. and, when printing, follow existing patterns (note there are no prints of raw char[]).